src/options.cc

```diff
@@ -67,6 +70,29 @@ bool Options::addDefaultHandler(std::shared_ptr<NginxConfigStatement> handler_co
     return addHandler(type, path, *handler_config->child_block_);
 }
 
+bool Options::addAuthentication(std::shared_ptr<NginxConfigStatement> auth_config) {
+    if (auth_config->tokens_.size() != REALM_SIZE) {
+        std::cerr << "A path has not been specified for the realm.\n";
+        return false;
+    }
+
+    auth = new Authentication();
+    std::map<std::string, std::string> AuthMap;
+    for (auto& statements : auth_config->child_block_->statements_) {
+        if (statements->tokens_.size() != REALM_CRED_SIZE) {
+            std::cerr << "Not a valid credential statement. Need " << 
+                REALM_CRED_SIZE << " tokens.\n";
+            return false;
+        }
+        if(statements->tokens_[0] == "credential") {
```

Comment:
in the example config, you use "credentials" instead of "credential". Regardless, what do you think of "user" instead of "credentials"? a little less wordy maybe?